modules/formulize/class/durationElement.php

```diff
@@ -0,0 +1,271 @@
+		}
+
+		return $totalMinutes > 0 ? $totalMinutes : NULL;
+	}
+
+	function prepareDataForDataset($value, $handle, $entry_id) {
+		return $value;
+	}
+
+	// Convert stored minutes back to time units for display
+	function loadValue($value, $ele_value, $element)
+	{
+		if (!is_numeric($value)) {
+			return $ele_value;
+		}
+
+		$minutes = intval($value);
+		$breakdown = array();
+
+		// Convert minutes to time units
+		$remaining = $minutes;
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if ($ele_value['show_' . $unit]) {
+				$amount = floor($remaining / $multiplier);
+				$breakdown[$unit] = $amount;
+				$remaining -= ($amount * $multiplier);
+			}
+		}
+
+		$ele_value['values'] = $breakdown;
+		return $ele_value;
+	}
+
+	// Render the duration inputs
+	function render($ele_value, $caption, $markupName, $isDisabled, $element, $entry_id, $screen = false, $owner = null)
+	{
+		if ($isDisabled) {
+			// Render as text for disabled state
+			$output = "";
+			if (isset($ele_value['values'])) {
+				foreach ($ele_value['values'] as $unit => $amount) {
+					if ($amount > 0 && $ele_value['show_' . $unit]) {
+						$output .= $amount . " " . $unit . " ";
+					}
+				}
+			}
+			return new XoopsFormLabel($caption, trim($output), $markupName);
+		}
+
+		// Create container for inputs
+		$container = new XoopsFormElementTray($caption, ' ');
+
+		// Add input field for each enabled time unit
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			$unitMarkupName = $markupName . '[' . $unit . ']';
+			if ($ele_value['show_' . $unit]) {
+				$value = isset($ele_value['values'][$unit]) ? $ele_value['values'][$unit] : '';
+				${"input_$unit"} = new XoopsFormText(
+					ucfirst($unit).":",
+					$unitMarkupName,
+					$ele_value['size'],
+					5,
+					$value,
+					false,
+					true
+				);
+				${"input_$unit"}->setExtra("min='0'");
+				${"input_$unit"}->setExtra("class='formulize-duration-element-input numbers-only-textbox'");
+				${"input_$unit"}->setExtra(" onchange=\"javascript:formulizechanged=1;\" jquerytag=\"$unitMarkupName\" ");
+				$container->addElement(${"input_$unit"});
+			}
+		}
+
+		$renderedElement = $container->render();
+
+		$form_ele = new XoopsFormLabel(
+			$caption,
+			trans($renderedElement),
+			$markupName
+		);
+
+		return $form_ele;
+	}
+
+	// Validate the duration is within bounds
+	function generateValidationCode($caption, $markupName, $element, $entry_id = false)
+	{
+		$validationCode = array();
+		$ele_value = $element->getVar('ele_value');
+
+		// Basic required validation if element is required
+		if ($element->getVar('ele_req')) {
+			$validationCode[] = "var hasValue = false;\n";
+			foreach ($this->timeUnits as $unit => $multiplier) {
+				if ($ele_value['show_' . $unit]) {
+					$validationCode[] = "if(myform['{$markupName}[{$unit}]'].value != '') hasValue = true;\n";
+				}
+			}
+			$validationCode[] = "if(!hasValue) {\n";
+			$validationCode[] = "    window.alert('Please enter a duration.');\n";
```

Comment:
In this and the other alerts to the user, I suggest using the caption of the element. Better user experience to direct them to where the problem is. I think we can also set focus on the element via the $markupName which is definitely a best practice too.